### ISSUE TITLE
Add "compile" animation to the run button

### ISSUE THREAD

**Issue by processing-bugs** [processing/processing#136]
_Original author: b...@processing.org (June 07, 2010 01:03:41)_

This bug automatically added from:
http://dev.processing.org/bugs/show_bug.cgi?id=776

Comment from fry, 2008-05-28 13:09

There can be a (sometimes significant) lag after clicking Run and seeing it
turn yellow. This compile and VM startup lag should instead be animated
using a spinner to make it clearer to users that things are happening.

_Original issue: http://code.google.com/p/processing/issues/detail?id=97_

**Comment by ndeleon** [comment]
Hey, I was wondering if this issue is still relevant, this could really help new users to understand how Processing works internally. I would be happy to go go about figuring out some way to make this happen.

**Comment by benfry** [comment]
Yeah, this is an old one—not sure of the current priority with the GUI changing. Any thoughts, @REAS or @shiffman?

**Comment by shiffman** [comment]
I have not noticed this to be an issue, but don't see any downside if it's integrated nicely with the lovely work James has been doing.

**Comment by REAS** [comment]
I like the idea. I'm not sure how difficult it is to implement.

**Comment by iamutkarshtiwari** [comment]
@benfry Is there any way to add delay to the compilation time? I want to test my patch for this enhancement.

**Comment by iamutkarshtiwari** [comment]
Here is the PR to add this requested feature. https://github.com/processing/processing/pull/4316

Waiting for reviews. :)

**Comment by damcoolkunal** [comment]
import android.view.View;
import android.view.animation.Animation;
import android.view.animation.Transformation;

public class HeightAnimation extends Animation {
    protected final int originalHeight;
    protected final View view;
    protected float perValue;

    public HeightAnimation(View view, int fromHeight, int toHeight) {
        this.view = view;
        this.originalHeight = fromHeight;
        this.perValue = (toHeight - fromHeight);
    }

    @Override
    protected void applyTransformation(float interpolatedTime, Transformation t) {
        view.getLayoutParams().height = (int) (originalHeight + perValue * interpolatedTime);
        view.requestLayout();
    }

    @Override
    public boolean willChangeBounds() {
        return true;
    }

**Comment by benfry** [comment]
Removed off-topic comment and locking this to prevent further spam.
